### PR TITLE
Port cub::DeviceScan tests to catch2

### PR DESCRIPTION
**Pull request by elstehle**
## Description

This PR partially addresses https://github.com/NVIDIA/cccl/issues/85. It migrates tests for `DeviceSegmentedReduce`, i.e., segmented-problem reduction algorithms, to Catch2. 

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cub/test/catch2_test_device_scan.cu [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  GEN_TYPE_RANDOM,
+  /// Constant value as input data
+  GEN_TYPE_CONST
+};
+
+CUB_TEST("Device scan works with all device interfaces", "[scan][device]", full_type_list)
+{
+  using params   = params_t<TestType>;
+  using item_t   = typename params::item_t;
+  using output_t = typename params::output_t;
```
Comment (senior-zero) [inline cub/test/catch2_test_device_scan.cu:96]: I'd prefer `input_t` / `output_t`.

File: cub/test/catch2_test_device_scan_iterators.cu [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    device_exclusive_sum(in_it, d_out_it, num_items);
+
+    // Verify result
+    REQUIRE(expected_result == out_result);
+  }
+
+  SECTION("inclusive scan")
+  {
+    using op_t    = cub::Min;
+    using accum_t = cub::detail::accumulator_t<op_t, item_t, item_t>;
```
Comment (senior-zero) [inline cub/test/catch2_test_device_scan_iterators.cu:124]: Seems to be unused

File: cub/test/catch2_test_device_scan_iterators.cu [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+    // Prepare verification data
+    thrust::host_vector<output_t> expected_result(num_items);
+    std::inclusive_scan(in_it,
+                        in_it + num_items,
+                        expected_result.begin(),
+                        op_t{},
+                        cub::NumericTraits<item_t>::Max());
```
Comment (senior-zero) [inline cub/test/catch2_test_device_scan_iterators.cu:132]: Did you mean `cub::NumericTraits<accum_t>::Max()`?

File: cub/test/test_device_scan.cu [hunk 4]
```diff
@@ -1,1275 +0,0 @@
...
-  const int idx = static_cast<int>(threadIdx.x + blockIdx.x * blockDim.x);
-
-  if (idx < num_items)
-  {
-    d_in[idx] = CustomInputT(1);
-    d_out[idx] = CustomOutputT{d_ok_count, idx};
-  }
-}
-
-void TestAccumulatorTypes()
```
Comment (senior-zero) [inline cub/test/test_device_scan.cu:1124]: Is there an alternative to `TestAccumulatorTypes` in the new tests?

**Review by senior-zero — APPROVED** [review]
A few minor comments below.

File: cub/test/catch2_test_device_scan.cu [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+#if TEST_TYPES != 3
+  SECTION("inclusive sum")
+  {
+    using op_t    = cub::Sum;
+    using accum_t = cub::detail::accumulator_t<op_t, item_t, item_t>;
+
+    // Prepare verification data
+    thrust::host_vector<item_t> host_items(in_items);
+    thrust::host_vector<output_t> expected_result(num_items);
+    std::inclusive_scan(host_items.cbegin(),
```
Comment (senior-zero) [inline cub/test/catch2_test_device_scan.cu:139]: Seems to require C++17

File: cub/test/test_device_scan.cu (hunk 4 above)
Comment (elstehle) [reply cub/test/test_device_scan.cu:1124]: Thanks! I've added a test that resembles `TestAccumulatorTypes` to `catch2_test_device_scan_iterators.cu`

**Pull request merged**